server/jetstream_cluster.go

```diff
@@ -7944,9 +7964,24 @@ func (mset *stream) processClusteredInboundMsg(subject, reply string, hdr, msg [
 			if state.Bytes+bytesPending > uint64(maxBytes) {
 				err = ErrMaxBytes
 			}
+		} else if maxMsgsPer > 0 && discardNewPerSubject {
+			totals := mset.store.SubjectsTotals(subject)
```

Comment:
This could be very large.. We are getting this each time?

Reply:
Only for wq/interest + discard new + discard new per subject streams.

Reply:
This can be very slow based on the shape of the stream. This concerns me. As does tracking inflight.